uuid.go

```diff
@@ -344,6 +345,30 @@ func (u *NullUUID) Scan(src interface{}) error {
 	return u.UUID.Scan(src)
 }
 
+// MarshalJSON marshalls the NullUUID as nil or the nested UUID
+func (u NullUUID) MarshalJSON() ([]byte, error) {
+	if u.Valid == false {
+		return json.Marshal(nil)
+	}
+	return json.Marshal(u.UUID)
+}
+
+// UnmarshalJSON unmarshalls a NullUUID
+func (u *NullUUID) UnmarshalJSON(b []byte) error {
+	var s *UUID
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	if s != nil {
+		u.Valid = true
+		u.UUID = *s
+	} else {
+		u.Valid = false
+	}
+	return nil
+}
```

Comment:
> @blastrock JSON decoder tokenizes input and it will pass "null" to UnmarshalJSON in any way.

Indeed, it seems so. But then ```u.UnmarshalJson("   null ")``` wouldn't work, but it may be acceptable. Just tell me your decision on that and I'll update the PR.

> If you look closer besides stylistic changes I removed intermediate UUID allocation by passing &u.UUID to json.Unmarshal.

Right, I didn't notice that, and I don't see any other way to remove that allocation.

> I agree with @shurcooL's comment about initializing all fields but it might be Scan method does it the most correct way by initializing UUID struct member to be Nil too.

Will do.